src/main/java/hudson/plugins/mercurial/traits/CleanMercurialSCMSourceTrait.java

```diff
@@ -0,0 +1,29 @@
+package hudson.plugins.mercurial.traits;
+
+import hudson.Extension;
+import hudson.plugins.mercurial.MercurialSCMBuilder;
+import hudson.scm.SCM;
+import javax.annotation.Nonnull;
+import jenkins.scm.api.trait.SCMBuilder;
+import jenkins.scm.api.trait.SCMSourceTrait;
+import jenkins.scm.api.trait.SCMSourceTraitDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class CleanMercurialSCMSourceTrait extends SCMSourceTrait {
+    @DataBoundConstructor public CleanMercurialSCMSourceTrait() { }
+
+    @Override protected <B extends SCMBuilder<B, S>, S extends SCM> void decorateBuilder(B builder) {
+        ((MercurialSCMBuilder<?>) builder).withClean(true);
```

Comment:
Why the exotic typing of the method signature if we are going to wind up casting anyway?